rust/src/bls/api.rs

```diff
@@ -181,39 +166,35 @@ pub unsafe extern "C" fn fil_verify(
 ///
 /// # Arguments
 ///
-/// * `signature_ptr`             - pointer to a signature byte array (SIGNATURE_BYTES long)
-/// * `messages_ptr`              - pointer to an array containing the pointers to the messages
+/// * `signature`             - pointer to a signature byte array (SIGNATURE_BYTES long)
+/// * `messages`              - pointer to an array containing the pointers to the messages
 /// * `messages_sizes_ptr`        - pointer to an array containing the lengths of the messages
 /// * `messages_len`              - length of the two messages arrays
 /// * `flattened_public_keys_ptr` - pointer to a byte array containing public keys
 /// * `flattened_public_keys_len` - length of the array
-#[no_mangle]
-pub unsafe extern "C" fn fil_hash_verify(
-    signature_ptr: *const u8,
-    flattened_messages_ptr: *const u8,
-    flattened_messages_len: libc::size_t,
-    message_sizes_ptr: *const libc::size_t,
-    message_sizes_len: libc::size_t,
-    flattened_public_keys_ptr: *const u8,
-    flattened_public_keys_len: libc::size_t,
+#[ffi_export]
+pub fn hash_verify(
+    signature: c_slice::Ref<u8>,
+    flattened_messages: c_slice::Ref<u8>,
+    message_sizes: c_slice::Ref<libc::size_t>,
+    flattened_public_keys: c_slice::Ref<u8>,
 ) -> libc::c_int {
     // prep request
-    let raw_signature = from_raw_parts(signature_ptr, SIGNATURE_BYTES);
-    let signature = try_ffi!(Signature::from_bytes(raw_signature), 0);
+    let signature = try_ffi!(Signature::from_bytes(&signature), 0);
 
-    let flattened = from_raw_parts(flattened_messages_ptr, flattened_messages_len);
-    let chunk_sizes = from_raw_parts(message_sizes_ptr, message_sizes_len);
+    let flattened = flattened_messages;
+    let chunk_sizes = message_sizes;
```

Comment:
I'd remove those, unless you think naming it `chunk_sizes` makes things clearer (I don't think so).

Reply:
fixed